controllers/component_image_controller.go

```diff
@@ -187,119 +145,76 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
-			quayClient := r.BuildQuayClient(log)
-			repo, pushRobotAccount, pullRobotAccount, err := r.generateImageRepository(ctx, quayClient, component, requestRepositoryOpts)
-			if err != nil {
-				if err.Error() == "payment required" {
-					log.Info("failed to create private image repository due to quay plan limit", l.Audit, "true")
-					repositoryInfo.Message = "Quay organization plan doesn't allow private image repositories"
-				} else {
-					log.Error(err, "Error in the repository generation process", l.Audit, "true")
-					return ctrl.Result{}, r.reportError(ctx, component, "failed to generate image repository")
-				}
-			} else {
-				if repo == nil || pushRobotAccount == nil || pullRobotAccount == nil {
-					log.Error(nil, "Unknown error in the repository generation process", l.Audit, "true")
-					return ctrl.Result{}, r.reportError(ctx, component, "failed to generate image repository: unknown error")
-				}
-				log.Info(fmt.Sprintf("Prepared image repository %s for Component", repo.Name), l.Action, l.ActionAdd)
-
-				imageURL := fmt.Sprintf("quay.io/%s/%s", r.QuayOrganization, repo.Name)
-
-				// Create secrets with the repository credentials
-				pushSecretName := component.Name
-				_, err := r.ensureRobotAccountSecret(ctx, component, pushRobotAccount, pushSecretName, imageURL)
-				if err != nil {
-					return ctrl.Result{}, err
-				}
-				log.Info(fmt.Sprintf("Prepared image registry push secret %s for Component", pushRobotAccount.Name), l.Action, l.ActionUpdate)
-
-				// Propagate the pull secret into all environments
-				pullSecretName := pushSecretName + "-pull"
-				if err := r.ensureComponentPullSecret(ctx, component, pullSecretName, pullRobotAccount, imageURL); err != nil {
-					return ctrl.Result{}, err
-				}
-				log.Info(fmt.Sprintf("Prepared remote secret %s for Component", pullSecretName), l.Action, l.ActionUpdate)
-
-				// Prepare data to update the component with
-				repositoryInfo = ImageRepositoryStatus{
-					Image:      imageURL,
-					Visibility: requestRepositoryOpts.Visibility,
-					Secret:     pushSecretName,
-				}
+	imageRepositoryFound := ""
+	for _, imageRepository := range imageRepositoriesList.Items {
+		for _, owner := range imageRepository.ObjectMeta.OwnerReferences {
+			if owner.UID == component.UID {
+				imageRepositoryFound = imageRepository.Name
+				break
 			}
 		}
 	}
-	repositoryInfoBytes, _ := json.Marshal(repositoryInfo)
 
-	// Update component with the generated data and add finalizer
-	err = r.Client.Get(ctx, req.NamespacedName, component)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("error reading component: %w", err)
-	}
-	if component.ObjectMeta.DeletionTimestamp.IsZero() {
-		component.Annotations[ImageAnnotationName] = string(repositoryInfoBytes)
-		delete(component.Annotations, GenerateImageAnnotationName)
+	if imageRepositoryFound != "" {
+		log.Info("Image repository already exists", "ImageRepositoryName", imageRepositoryFound)
```

Comment:
Do we log related component anywhere?

Reply:
bellow when component is updated, but I can add it to this log as well